### PR TITLE
chainhook 1.5.1 (new formula)

### PR DESCRIPTION
**Pull request by MicaiahReid**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/chainhook.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Chainhook < Formula
+  desc "Reorg-aware indexing engine for the Stacks & Bitcoin blockchains"
+  homepage "https://github.com/hirosystems/chainhook"
+  # pull from git tag to get submodules
```
Comment (chenrui333) [inline Formula/c/chainhook.rb:4]: what is this for?

File: Formula/c/chainhook.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      tag:      "v1.4.1",
+      revision: "9d9f0d66dc32c7680fdabd3dba2ecf2847dd2b2a"
+  license "GPL-3.0-only"
+  head "https://github.com/hirosystems/chainhook.git", branch: "develop"
+
+  livecheck do
+    url :stable
+    regex(/^v?(\d+(?:\.\d+)+)$/i)
+  end
+
```
Comment (chenrui333) [inline Formula/c/chainhook.rb:15]:
```suggestion
```

File: Formula/c/chainhook.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class Chainhook < Formula
+  desc "Reorg-aware indexing engine for the Stacks & Bitcoin blockchains"
+  homepage "https://github.com/hirosystems/chainhook"
+  url "https://github.com/hirosystems/chainhook.git",
+      tag:      "v1.4.1",
+      revision: "9d9f0d66dc32c7680fdabd3dba2ecf2847dd2b2a"
```
Comment (chenrui333) [inline Formula/c/chainhook.rb:6]: please use tarball

Comment (MicaiahReid) [reply Formula/c/chainhook.rb:6]: Hi, thanks for the feedback @chenrui333. 

Can I ask why we need to use a tarball? I used another comparable rust formula as a template (https://github.com/Homebrew/homebrew-core/blob/master/Formula/c/clarinet.rb), and it builds from a git revision as well.

File: Formula/c/chainhook.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  license "GPL-3.0-only"
+  head "https://github.com/hirosystems/chainhook.git", branch: "develop"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "chainhook-install", "--root", prefix.to_s
```
Comment (chenrui333) [inline Formula/c/chainhook.rb:13]: also use std_cargo_args in here as well, see some build change in https://github.com/Homebrew/homebrew-core/pull/168061

**Comment by SMillerDev** [comment]
>     Unable to find libclang: "couldn't find any valid shared libraries matching: ['libclang.so', 'libclang-*.so', 'libclang.so.*', 'libclang-*.so.*'], set the `LIBCLANG_PATH` environment variable to a path where one of these files can be found (invalid: [])"

**Comment by MicaiahReid** [comment]
@chenrui333 @SMillerDev I have addressed your comments!

I thought I fixed all of the commit messages to match the expected style, but it still is failing CI on that point.

File: Formula/c/chainhook.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  url "https://github.com/hirosystems/chainhook/archive/refs/tags/v1.5.1.tar.gz"
+  sha256 "ccea19c9e81672ddb58cfda84cdd5d923cc26af422c2c1b81d9424125d85ba2d"
+  license "GPL-3.0-only"
+  head "https://github.com/hirosystems/chainhook.git", branch: "develop"
+
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "llvm@15" => :build # for libclang
+  end
```
Comment (chenrui333) [inline Formula/c/chainhook.rb:13]: we can do something like this?
```suggestion
  uses_from_macos "llvm" => :build # for libclang
```

File: Formula/c/chainhook.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  def install
+    system "cargo", "install", "--force", "--features", "cli", "--features",
+     "debug", "--no-default-features", *std_cargo_args(path: "components/chainhook-cli")
```
[the remark this answers is not in the document]
Comment (MicaiahReid) [reply Formula/c/chainhook.rb:17]: This is really just poor naming of the feature on the Chainhook side. This is how we recommend users install Chainhook.

**Comment by MicaiahReid** [comment]
@chenrui333 Could I get a re-review on this?

File: Formula/c/chainhook.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  license "GPL-3.0-only"
+  head "https://github.com/hirosystems/chainhook.git", branch: "develop"
+
+  depends_on "rust" => :build
+
+  uses_from_macos "llvm" => :build # for libclang
+
+  def install
+    system "cargo", "install", "--force", "--features", "cli", "--features",
+     "debug", "--no-default-features", *std_cargo_args(path: "components/chainhook-cli")
```
Comment (chenrui333) [inline Formula/c/chainhook.rb:15]:
```suggestion
    system "cargo", "install", "--features", "cli,debug", "--no-default-features", 
                                *std_cargo_args(path: "components/chainhook-cli")
```

File: Formula/c/chainhook.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/c/chainhook.rb:15]: any reference about that `debug` feature?

**Comment by chenrui333** [comment]
@MicaiahReid one last comment, can you also squash all your commits as well? And add this formula into autobump.txt file. Thanks!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@MicaiahReid, thanks for your first contribution to homebrew-core!! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**